platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/creation/EventCreationManager.java

```diff
@@ -134,11 +135,12 @@ public void registerEvent(@NonNull final GossipEvent event) {
     }
 
     /**
-     * Update the minimum generation non-ancient.
+     * Update the non-ancient event window, defining the minimum threshold for an event to be non-ancient.
      *
-     * @param minimumGenerationNonAncient the new minimum generation non-ancient
+     * @param nonAncientEventWindow the non-ancient event window
      */
-    public void setMinimumGenerationNonAncient(final long minimumGenerationNonAncient) {
-        creator.setMinimumGenerationNonAncient(minimumGenerationNonAncient);
+    public void setNonAncientEventWindow(@NonNull final NonAncientEventWindow nonAncientEventWindow) {
+        // FUTURE WORK: change from minGenNonAncient to minRoundNonAncient
+        creator.setMinimumGenerationNonAncient(nonAncientEventWindow.minGenNonAncient());
```

Comment:
I think that this is not accurate. Event creation manager should actually start storing `NonAncientEventWindow` and use the `isAncient` method. That way, we can flip a switch and switch the whole system at once.

Reply:
The event creation manager should not be aware what is used for defining ancient.

Reply:
Event creation manager internally uses a sequence map, so an `isAncient()` method will be insufficient for its needs.

That being said, I think it would be a good idea to pass in the `NonAncientEventWindow` to all classes that are accepting a minimum generation. That way the API can be formalized and completed even before we throw the switch between generations and birth rounds.

Reply:
I'll update more of the API in this PR.  We don't throw the switch until the end of the `Green Wire` task list, so the API might change further in the other PRs that implement the component individual changes. https://github.com/hashgraph/hedera-services/issues/5307

Reply:
My comment was mostly about the `FUTURE WORK` comment. I don't think we want to flip the switch in each individual component. So maybe the comment should change

Reply:
I've updated more of the API and changed the code to use `NonAncientEventWindow.isAncient()` instead of a comparison to the `minimumGenerationNonAncient` value.